features/netsocket/nsapi_ppp.h

```diff
@@ -0,0 +1,34 @@
+/** \addtogroup netsocket */
+/** @{*/
+/* nsapi_ppp.h
+ * Modified work Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef NSAPI_PPP_H_
+#define NSAPI_PPP_H_
+
+#include "FileHandle.h"
+#include "NetworkStack.h"
+
+namespace mbed {
+
+NetworkStack *nsapi_ppp_get_stack();
```

Comment:
mbed OS does require that applications pass in pins, but as requiring pin knowledge inherently makes applications totally non-portable, the convention seems to be that there are standard virtual names for pins like the debug UART and Arduino header pins, so applications can try passing those. This are such standard virtual pin names for the modem UART.

If they didn't exist, that would be another 5 lines of ifdef or JSON per platform in easy-connect or any application trying to use ReferenceCellularInterface.

(I am feeling that it should be slightly more specific than DEVICE_MODEM - the pins would be specifically for a UART-attached modem, when it could be USB or SPI.

DEVICE_MODEM could cover the common auxiliary modem_api.h, and DEVICE_MODEM_UART could say it's  using a UART data connection. The fact we use enums rather than defines is unfortunate - means we can't do `#ifdef MDMTX`, and need some define to say that pin names are available).